auth/role.go

```diff
@@ -32,6 +33,10 @@ type roleImpl struct {
 	Deleted           bool            `json:"deleted,omitempty"`
 	vbNo              *uint16
 	cas               uint64
+
+	// warnChanThresholdOnce ensures that the check for channels
+	// per user threshold is only performed exactly once.
+	warnChanThresholdOnce sync.Once
```

Comment:
Why is this on `roleImpl` instead of `userImpl`?

Reply:
I agree, it should be on userImpl.